api/files/serializers.py

```diff
@@ -438,3 +454,22 @@ def get_file_download_link(obj, version=None, view_only=None):
     if view_only:
         url.args['view_only'] = view_only
     return url.url
+
+
+def get_file_render_link(mfr_url, download_url, version=None):
+    download_url_args = {
+        'direct': None,
+        'mode': 'render',
+    }
+    if version:
+        download_url_args['revision'] = version
+
+    render_url = furl.furl(mfr_url).set(
+        path=['render'],
+        args={
+            'url': furl.furl(download_url).set(
+                args=download_url_args,
+            ),
+        },
+    )
+    return render_url.url
```

Comment:
This returns the same type of link we use to render preprint files:
https://mfr.us.staging.osf.io/render?url=https://staging.osf.io/download/abcde/?direct&mode=render

It differs from the render link we use to render projects, but I double checked on staging, and both forms work for projects (main different is that download links are different, one has "download" as part of the URL and another uses an action query param with the value as "download".  So again, I render the form we use for preprints, and it works for projects as well.

https://mfr.us.staging.osf.io/render?url=https://staging.osf.io/n92x6/?action=download&mode=render&direct&public_file=False&initialWidth=389&childId=mfrIframe&parentTitle=OSF | hello.txt&parentUrl=https://staging.osf.io/n92x6/